### PR TITLE
EOS-28031: Support bundle enhancement for adding time filter

### PR DESCRIPTION
**Pull request by abhijit1patil**
Signed-off-by: abhijit1patil <abhijit.1.patil@seagate.com>

# Problem Statement
-[EOS-28031](https://jts.seagate.com/browse/EOS-28031) - Support bundle enhancement for adding time filter

# Design
    #NOTE: Below are the additional arguments, that all component specific Support Bundle script should accept.
    # --duration -> ISO 8061 format, "P" separates, the start datetime and duration to be considered after that. "T" separated date and time. Default is P5d.

# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Testing 
Testcase 1 (Only Duration Filter): - csm agent logs are filtered based on the Duration filter
eos28392-1

TestCase 2: Support bundle with Size and Duration Filter
eos28392-2

Testcase 3: Support Bundle generation for Cortex
eos28392-3

  Checklist for Author
- [x] Unit and System Tests are added
- [ ] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [ ] Testing was performed with RPM

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [ ] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [x] JIRA number/GitHub Issue added to PR
- [x] PR is self reviewed
- [ ] Jira and state/status is updated and JIRA is updated with PR link
- [ ] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

**Comment by cortx-admin** [comment]
Can one of the admins verify this patch?

File: csm/cli/support_bundle/csm_bundle_generate.py [hunk 1]
```diff
@@ -139,25 +139,25 @@ def generate_bundle(args):
...
         os.makedirs(csm_size_filtered_logs_dir,exist_ok=True)
-        csm_size_time_filtered_logs_dir = os.path.join(const.CSM_SETUP_LOG_DIR,
-            f"{const.CSM_COMPONENT_NAME}_logs_size_time")
-        GenerateCsmBundle.__clear_tmp_files(csm_size_time_filtered_logs_dir)
-        os.makedirs(csm_size_time_filtered_logs_dir,exist_ok=True)
-        FilterLog.limit_size(csm_log_path, csm_size_filtered_logs_dir,
-            GenerateCsmBundle.size_limit, const.CSM_COMPONENT_NAME)
+        
+        csm_time_filtered_logs_dir = os.path.join(const.CSM_SETUP_LOG_DIR,
+            f"{const.CSM_COMPONENT_NAME}_logs_")
```
Comment (rohitkolapkar) [inline csm/cli/support_bundle/csm_bundle_generate.py:147]: does csm_size_filtered_logs_dir and csm_time_filtered_logs_dir expected to have same path i.e. "/tmp/csm_logs_" ?

Comment (abhijit1patil) [reply csm/cli/support_bundle/csm_bundle_generate.py:147]: I have checked the code for utils.. It is better to keep to different path.. I will modify the code

File: csm/cli/support_bundle/csm_bundle_generate.py [hunk 2]
```diff
@@ -137,27 +137,27 @@ def generate_bundle(args):
         GenerateCsmBundle.stacktrace = args[const.SB_STACKTRACE]
         target_path = os.path.join(GenerateCsmBundle.target_path, const.CSM_COMPONENT_NAME)
         os.makedirs(target_path,exist_ok=True)
-
+        # Apply Time filter on CSM Logs Default: P5d
+        csm_time_filtered_logs_dir = os.path.join(const.CSM_SETUP_LOG_DIR,
+            f"{const.CSM_COMPONENT_NAME}_logs_time")
```
Comment (pawankumarsrivastava) [inline csm/cli/support_bundle/csm_bundle_generate.py:142]: 1. Create 1 folder name /tmp/CSM_filtered_logs
2. Do time filtering. 
3. Created a new folder /tmp/CSM_filtered_logs/size
4. Input folder step 1 and output folder step 3

**Review by pawankumarsrivastava — CHANGES_REQUESTED** [review]

**Comment by abhijit1patil** [comment]
Testcase 1 (Only Duration Filter): - csm agent logs are filtered based on the Duration filter
eos28392-1

**Comment by abhijit1patil** [comment]
TestCase 2: Support bundle with Size and Duration Filter
eos28392-2

**Comment by abhijit1patil** [comment]
Testcase 3: Support Bundle generation for Cortex
eos28392-3

**Review by pawankumarsrivastava — APPROVED** [review]

**Review by rohitkolapkar — APPROVED** [review]

**Comment by pawankumarsrivastava** [comment]
ok to test

**Comment by pawankumarsrivastava** [comment]
run mini-provisioning

**Comment by pawankumarsrivastava** [comment]
ok to test

**Comment by kupranay** [comment]
retest this please

**Pull request merged**